src/main.rs

```diff
@@ -1,3 +1,176 @@
+use core::time::Duration;
+use serialport::{available_ports, SerialPort};
+use std::io::Read;
+
+static INSTRUCTIONS_LM: &str = "Linux Module (probably) faulty, return to UniElec";
+static INSTRUCTIONS_BUTTON: &str = "Check button";
+
+fn open_serial_port(path: &str) -> Result<Box<dyn SerialPort>, Box<dyn std::error::Error>> {
+    Ok(serialport::new(path, 115_200)
+        .timeout(Duration::from_millis(100))
+        .open()?)
+}
+
+fn remove_non_printable(s: &str) -> String {
+    s.chars()
+        .filter(|&c| c.is_ascii_graphic() || c.is_ascii_whitespace())
```

Comment:
this doesn't seem sufficient, e.g. what about `is_ascii_alphanumeric()`?

Reply:
How so?

> Checks if the value is an ASCII graphic character: U+0021 ‘!’ ..= U+007E ‘~’.

+ whitespace is exactly what I want.